exercises/robot-simulator/canonical-data.json

```diff
@@ -294,10 +312,14 @@
       ]
     },
     {
-      "description": "Where R = Turn Right, L = Turn Left and A = Advance, the robot can follow a series of instructions and end up with the correct position and direction",
+      "description": "Series of instructions",
+      "comments": [ "The robot can follow a series of instructions and end up with the correct position and direction.",
+                    "Where R = Turn Right, L = Turn Left and A = Advance"
+      ]
```

Comment:
Pharo generates individual method names for each unit test to be unique and fully descriptive by cascading the layers of descriptions. This works for all but a few exercises like robot-simulator, for which methods names arise as follows...
```
test01_ARobotIsCreatedWithAPositionAndADirectionRobotsAreCreatedWithAPositionAndDirection
test02_ARobotIsCreatedWithAPositionAndADirectionNegativePositionsAreAllowed
test03_RotatesTheRobotsDirection90DegreesClockwiseChangesTheDirectionFromNorthToEast
test04_RotatesTheRobotsDirection90DegreesClockwiseChangesTheDirectionFromEastToSouth
test05_RotatesTheRobotsDirection90DegreesClockwiseChangesTheDirectionFromSouthToWest
test06_RotatesTheRobotsDirection90DegreesClockwiseChangesTheDirectionFromWestToNorth
test07_RotatesTheRobotsDirection90DegreesCounterclockwiseChangesTheDirectionFromNorthToWest
test08_RotatesTheRobotsDirection90DegreesCounterclockwiseChangesTheDirectionFromWestToSouth
test09_RotatesTheRobotsDirection90DegreesCounterclockwiseChangesTheDirectionFromSouthToEast
test10_RotatesTheRobotsDirection90DegreesCounterclockwiseChangesTheDirectionFromEastToNorth
test11_MovesTheRobotForward1SpaceInTheDirectionItIsPointingIncreasesTheYCoordinateOneWhenFacingNorth
test12_MovesTheRobotForward1SpaceInTheDirectionItIsPointingDecreasesTheYCoordinateByOneWhenFacingSouth
test13_MovesTheRobotForward1SpaceInTheDirectionItIsPointingIncreasesTheXCoordinateByOneWhenFacingEast
test14_MovesTheRobotForward1SpaceInTheDirectionItIsPointingDecreasesTheXCoordinateByOneWhenFacingWest
test15_WhereRTurnRightLTurnLeftAndAAdvanceTheRobotCanFollowASeriesOfInstructionsAndEndUpWithTheCorrectPositionAndDirectionInstructionsToMoveEastAndNorthFromREADME
test16_WhereRTurnRightLTurnLeftAndAAdvanceTheRobotCanFollowASeriesOfInstructionsAndEndUpWithTheCorrectPositionAndDirectionInstructionsToMoveWestAndNorth
test17_WhereRTurnRightLTurnLeftAndAAdvanceTheRobotCanFollowASeriesOfInstructionsAndEndUpWithTheCorrectPositionAndDirectionInstructionsToMoveWestAndSouth
test18_WhereRTurnRightLTurnLeftAndAAdvanceTheRobotCanFollowASeriesOfInstructionsAndEndUpWithTheCorrectPositionAndDirectionInstructionsToMoveEastAndNorth
```
which is a bit awful to ask a GUI IDE to deal with.

The [canonical schema](https://github.com/exercism/problem-specifications/blob/master/canonical-schema.json) specifies both  "description" and "comments".  One of those is implicitly longer, and the other needs to provide a **minimal identifier** for the test.  What other data is available to generate unit test method names of reasonable length?
btw, How are method names generated for other languages? 

I didn't spend the time making this edit to shorten the descriptions only for the sake of making them shorter.  Its about distinguishing a descriptive-identifier from a verbose-comment.
I'd previously suggested in #1473 adding a separate field for this, but the preferred path given to me was shortening the "description" field.  I'd be interested in review from @rpottsoh @ErikSchierboom.

Reply:
> single word or 3 word descriptions. You loose a lot of information in the test suite this way.

That is what comments are for.  
But I appreciate that there is a random sampling of approaches among language tracks, and curious on the details of how "comments" are handled.

Reply:
> which is a bit awful to ask a GUI IDE to deal with.

Oof. I can imagine. That looks like quite a bit 😓. Since the names are pretty descriptive in their original form, would this work by **not** cascading them?

> and curious on the details of how "comments" are handled.

Comments are for the generator / implementer, free form and no rules.

Reply:
> How are method names generated for other languages?

#1411 has some info on that. To add to it: If we look at Python, Go and C# implementations of this exercise, only C# uses the description for method naming. Perhaps you could take a look at those for other options.
In the WIP Julia generator I'm working on the description is added as a comment and the comment is discarded but the idea of naming (or describing) every single test is not a thing in the language anyway.

> That is what comments are for.

That's what the description is for. From the README:

> Keep in mind that the description should not simply explain what each case is (that is redundant information) but also why each case is there. For example, what kinds of implementation mistakes might this case help us find?

---

> I don't actually think there was anything wrong with _any_ of these descriptions.

I do think they don't do a great job of explaining the _why_ but these changes don't help.

Reply:
> I do think they don't do a great job of explaining the why but these changes don't help with that issue.

Right. What I meant that I didn't think the length was problematic as is ;)

Reply:
Since Python & Go don't use directly use the description, my changes will have no impact on them.  Can you provide some concrete code examples of where my changes cause harm?

Reviewing Python [robot_simulator_test.py](https://github.com/exercism/python/blob/master/exercises/robot-simulator/robot_simulator_test.py)  
and Go [robot_simulator_test.go](https://github.com/exercism/go/blob/master/exercises/robot-simulator/robot_simulator_test.go) [robot_simulator_step2_test.go](https://github.com/exercism/go/blob/master/exercises/robot-simulator/robot_simulator_step2_test.go) & [robot_simulator_step3_test.go](https://github.com/exercism/go/blob/master/exercises/robot-simulator/robot_simulator_step3_test.go) 
their method names look like they **aren't** generated and so don't **directly depend** on the descriptions.  So I don't understand what "other options" I can infer from them.  

Reviewing C#, it looks like their method names are generated similar to Pharo by concatenating each level's description.  Their longest is an 190 character identifier.  This change may be beneficial to them? @robkeim @GKotfis @ErikSchierboom @abecerramatias
```csharp
public void A_robot_is_created_with_a_position_and_a_direction_robots_are_created_with_a_position_and_direction()
public void A_robot_is_created_with_a_position_and_a_direction_negative_positions_are_allowed()
public void Rotates_the_robots_direction_90_degrees_clockwise_changes_the_direction_from_north_to_east()
public void Rotates_the_robots_direction_90_degrees_clockwise_changes_the_direction_from_east_to_south()
public void Rotates_the_robots_direction_90_degrees_clockwise_changes_the_direction_from_south_to_west()
public void Rotates_the_robots_direction_90_degrees_clockwise_changes_the_direction_from_west_to_north()
public void Rotates_the_robots_direction_90_degrees_counter_clockwise_changes_the_direction_from_north_to_west()
public void Rotates_the_robots_direction_90_degrees_counter_clockwise_changes_the_direction_from_west_to_south()
public void Rotates_the_robots_direction_90_degrees_counter_clockwise_changes_the_direction_from_south_to_east()
public void Rotates_the_robots_direction_90_degrees_counter_clockwise_changes_the_direction_from_east_to_north()
public void Moves_the_robot_forward_1_space_in_the_direction_it_is_pointing_increases_the_y_coordinate_one_when_facing_north()
public void Moves_the_robot_forward_1_space_in_the_direction_it_is_pointing_decreases_the_y_coordinate_by_one_when_facing_south()
public void Moves_the_robot_forward_1_space_in_the_direction_it_is_pointing_increases_the_x_coordinate_by_one_when_facing_east()
public void Moves_the_robot_forward_1_space_in_the_direction_it_is_pointing_decreases_the_x_coordinate_by_one_when_facing_west()
public void Where_r_turn_right_l_turn_left_and_a_advance_the_robot_can_follow_a_series_of_instructions_and_end_up_with_the_correct_position_and_direction_instructions_to_move_east_and_north_from_readme()
public void Where_r_turn_right_l_turn_left_and_a_advance_the_robot_can_follow_a_series_of_instructions_and_end_up_with_the_correct_position_and_direction_instructions_to_move_west_and_north()
public void Where_r_turn_right_l_turn_left_and_a_advance_the_robot_can_follow_a_series_of_instructions_and_end_up_with_the_correct_position_and_direction_instructions_to_move_west_and_south()
public void Where_r_turn_right_l_turn_left_and_a_advance_the_robot_can_follow_a_series_of_instructions_and_end_up_with_the_correct_position_and_direction_instructions_to_move_east_and_north()
```

Reply:
> Can you provide some concrete code examples of where my changes cause harm?

At least in the current form of the schema, it uses the description for something it's not intended to be, thus making the file less human-readable and more annoying to generate for generators that rely on the description being descriptive.

> So I don't understand what "other options" I can infer from them.

Ways to generate test suites without using the descriptions as method names directly.

Reply:
>At least in the current form of the schema, it uses the description for something it's not intended to be, thus making the file less human-readable and more annoying to generate for generators that rely on the description being descriptive.

@SaschaMann Please note, I'd prefer not to change these descriptions at all.  I'm only doing so [as I was directed to here](https://github.com/exercism/problem-specifications/issues/1473).